src/Universe/UniverseGenerator.php

```diff
@@ -0,0 +1,601 @@
+        $this->planetTypes = $planetTypes;
+        $this->cellRepo = $cellRepo;
+        $this->entityRepo = $entityRepo;
+        $this->starRepo = $starRepo;
+        $this->planetRepo = $planetRepo;
+        $this->asteroidsRepo = $asteroidsRepo;
+        $this->nebulaRepo = $nebulaRepo;
+        $this->wormholeRepo = $wormholeRepo;
+        $this->emptySpaceRepo = $emptySpaceRepo;
+        $this->databaseManager = $databaseManager;
+
+        $this->init();
+    }
+
+    private function init(): void
+    {
+        $this->sol_types = array_keys($this->solarTypes->getSolarTypeNames());
+        $this->planet_types = array_keys($this->planetTypes->getPlanetTypeNames());
+    }
+
+    public function exists(): bool
+    {
+        return $this->cellRepo->count() > 0;
+    }
+
+    public function isInhabited(): bool
+    {
+        return $this->planetRepo->countWithUser() > 0;
+    }
+
+    public function getSectorDimensions(): array
+    {
+        return $this->cellRepo->getSectorDimensions();
+    }
+
+    public function getCellDimensions(): array
+    {
+        return $this->cellRepo->getCellDimensions();
+    }
+
+    public function countStars(): int
+    {
+        return $this->starRepo->count();
+    }
+
+    public function countPlanets(): int
+    {
+        return $this->planetRepo->count();
+    }
+
+    public function countAsteroids(): int
+    {
+        return $this->asteroidsRepo->count();
+    }
+
+    public function countNebulas(): int
+    {
+        return $this->nebulaRepo->count();
+    }
+
+    public function countWormholes(): int
+    {
+        return $this->wormholeRepo->count();
+    }
+
+    public function countEmptySpace(): int
+    {
+        return $this->emptySpaceRepo->count();
+    }
+
+    /**
+     * Create the universe.
+     * And there was light!
+     */
+    public function create($mapImage = "", $mapPrecision = 95): void
+    {
+        $mtx = new Mutex();
+        $mtx->acquire();
+
+        $mapPrecision = max(0, $mapPrecision);
+        $mapPrecision = min($mapPrecision, 100);
+
+        echo "Lade Schöpfungs-Einstellungen...!<br>";
+        $sx_num = $this->config->param1Int('num_of_sectors');
+        $sy_num = $this->config->param2Int('num_of_sectors');
+        $cx_num = $this->config->param1Int('num_of_cells');
+        $cy_num = $this->config->param2Int('num_of_cells');
+        $perc_solsys = $this->config->getInt('space_percent_solsys');
+        $perc_asteroids = $this->config->getInt('space_percent_asteroids');
+        $perc_nebulas = $this->config->getInt('space_percent_nebulas');
+        $perc_wormholes = $this->config->getInt('space_percent_wormholes');
+
+        $sol_count = 0;
+        $nebula_count = 0;
+        $asteroids_count = 0;
+        $wormhole_count = 0;
+
+        echo "Erstelle Universum mit " . $sx_num * $sy_num . " Sektoren à " . $cx_num * $cy_num . " Zellen, d.h. " . $sx_num * $sy_num * $cx_num * $cy_num . " Zellen total<br>";
```

Comment:
Now that the code is moved to a class I don't think it should echo anymore. Instead we can return a list of steps or generation log which then gets rendered in the template/php script.

Reply:
You are completely right. Fixed in 
2015430